src/Kernel/IQSharpEngine.cs

```diff
@@ -342,7 +342,31 @@ public override async Task<ExecutionResult> ExecuteMundane(string input, IChanne
                 }
                 catch (CompilationErrorsException c)
                 {
-                    foreach (var m in c.Errors) channel.Stderr(m);
+                    // Check if the user likely tried to execute a magic
+                    // command and try to give a more helpful message in that
+                    // case.
+                    if (input.TrimStart().StartsWith("%") && input.Split("\n").Length == 1)
+                    {
+                        var attemptedMagic = input.Split(" ", 2)[0];
+                        var msg = $"No such magic command {attemptedMagic}.";
+                        if (MagicResolver is MagicSymbolResolver iqsResolver)
+                        {
+                            var similarMagic = iqsResolver
+                                .FindAllMagicSymbols()
+                                .Select(symbol =>
+                                    (symbol.Name, symbol.Name.EditDistanceFrom(attemptedMagic))
+                                )
+                                .OrderBy(pair => pair.Item2)
+                                .Take(3)
+                                .Select(symbol => symbol.Name);
+                            msg += $"\nPossibly similar magic commands:\n\t{string.Join(", ", similarMagic)}";
```

Comment:
would it be better to output each message individually (i.e. call `channel.Stderr` on each line), as opposed to a single concatenated string?

Reply:
Good point, let me go make that change, then.